_ext/asciidoctor_toc.rb

```diff
@@ -0,0 +1,18 @@
+require 'asciidoctor'
+
+module Awestruct::Extensions
+  class AsciidoctorTOC
+    def execute(site)
+      site.pages.each do |page|
+        if page.source_path.end_with? '.adoc'
+          document = Asciidoctor.load_file page.source_path, header_footer: true, safe: :safe, attributes: 'idprefix idseparator=-'
+          page.toc = true
```

Comment:
Maybe we could go further and store Asciidoctor:Document in page context ? https://github.com/awestruct/awestruct/issues/288